taurus/entity/object/tests/test_process_run.py

```diff
@@ -35,3 +40,20 @@ def test_ingredient_run():
 
     assert 'process' in repr(ingred_run)
     assert 'ingredients' in repr(proc_run)
+
+
+def test_invalid_assignment():
+    """Invalid assignments to `spec` throw a ValueError."""
+    with pytest.raises(ValueError):
+        ProcessRun("name", spec=[ProcessSpec("spec")])
```

Comment:
Shouldn't this and the similar exceptions be TypeErrors not ValueErrors?

Reply:
That's a good point. They should be TypeErrors. I suspect there are quite a few places where that should be changed. I'll do that.